contracts/src/v0.8/keystone/KeystoneRouter.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {IRouter} from "./interfaces/IRouter.sol";
+import {IReceiver} from "./interfaces/IReceiver.sol";
+import {OwnerIsCreator} from "../shared/access/OwnerIsCreator.sol";
+import {ITypeAndVersion} from "../shared/interfaces/ITypeAndVersion.sol";
+
+contract KeystoneRouter is IRouter, OwnerIsCreator, ITypeAndVersion {
+  error Unauthorized();
+
+  /// @notice This error is thrown whenever a message has already been processed.
+  /// @param messageId The ID of the message that was already processed
+  error AlreadyProcessed(bytes32 messageId);
+
+  event ForwarderAdded(address indexed forwarder);
+  event ForwarderRemoved(address indexed forwarder);
+
+  mapping(address forwarder => bool) internal s_forwarders;
```

Comment:
Feels like a good place to use EnumerableSet

Reply:
I think it would be overkill in this scenario.